File: 2900.longest-unequal-adjacent-groups-subsequence-i.go
// @leet start
func getLongestSubsequence(words []string, groups []int) []string {
  j := 0
  res := []string{words[j]}
  for i := 1; i < len(groups); i++ {
    if groups[i] != groups[j] {
      j = i
      res = append(res, words[j])
    }
  }
  return res
}
// @leet end

File: 2900.longest-unequal-adjacent-groups-subsequence-i.cs
// @leet start
public class Solution {
    public IList<string> GetLongestSubsequence(string[] words, int[] groups) {
      var j = 0;
      var res = new List<string>(){words[j]};
      for(int i = 1; i < words.Length; i++){
        if(groups[j] != groups[i]){
          j = i;
          res.Add(words[j]);
        }
      }
      return res;
    }
}
// @leet end
